src/lib/study-dfns.js

```diff
@@ -0,0 +1,478 @@
+      expected = null;
+      break;
+
+    case 'operation':
+      // Stringification behavior is typically defined with a
+      // "stringification behavior" definition scoped to the interface
+      if (idl.special === 'stringifier') {
+        expected.linkingText = ['stringification behavior', 'stringificationbehavior'];
+        expected.type = 'dfn';
+      }
+      // Ignore special "getter", "setter", "deleter" operations when they don't
+      // have an identifier. They should link to a definition in the prose, but
+      // the labels seem arbitrary for now.
+      // Also ignore default toJSON operations. Steps are defined in WebIDL.
+      else if ((idl.name ||
+            ((idl.special !== 'getter') &&
+            (idl.special !== 'setter') &&
+            (idl.special !== 'deleter'))) &&
+          !isDefaultToJSONOperation(idl)) {
+        expected.linkingText = [`${idl.name}(${serializeArgs(idl.arguments)})`];
+        expected.type = 'method';
+      }
+      else {
+        expected = null;
+      }
+      break;
+
+    case 'typedef':
+      break;
+
+    case 'argument':
+      expected = null;
+      break;
+
+    default:
+      console.warn('Unsupported IDL type', idl.type, idl);
+      expected = null;
+      break;
+  }
+
+  return expected;
+}
+
+
+/**
+ * Return the list of expected definitions from a parsed IDL extract entry.
+ *
+ * The function is recursive.
+ *
+ * @function
+ * @private
+ * @param {Object} idl The object that describes the IDL term in the
+ *   `idlparsed` extract.
+ * @return {Array} An array of expected definitions
+ */
+function getExpectedDfnsFromIdlDesc(idl, {excludeRoot} = {excludeRoot: false}) {
+  const res = [];
+  const parentIdl = idl;
+  const idlToProcess = excludeRoot ? [] : [idl];
+
+  switch (idl.type) {
+    case 'enum':
+      if (idl.values) {
+        idlToProcess.push(...idl.values);
+      }
+      break;
+
+    case 'callback':
+    case 'callback interface':
+    case 'dictionary':
+    case 'interface':
+    case 'interface mixin':
+    case 'namespace':
+      if (idl.members) {
+        idlToProcess.push(...idl.members);
+      }
+      break;
+  }
+
+  idlToProcess.forEach(idl => {
+    const expected = getExpectedDfnFromIdlDesc(idl, parentIdl);
+    if (expected) {
+      expected.access = 'public';
+      expected.informative = false;
+      res.push(expected);
+    }
+  });
+
+  return res;
+}
+
+
+/**
+ * Return true when the given IDL definition matches the expected definition.
+ *
+ * The function handles overloaded methods, though not properly. That is, it
+ * will only find the "right" definition for an overloaded method if the number
+ * and/or the name of the arguments differ between the overloaded definitions.
+ * Otherwise it will just match the first definition that looks good.
+ *
+ * The function works around Respec's issue #3200 for methods and constructors
+ * that take only optional parameters:
+ * https://github.com/w3c/respec/issues/3200
```

Comment:
```suggestion
 * https://github.com/speced/respec/issues/3200
```
(since that issue is fixed in ReSpec, I assume this remains useful for specs that were generated prior to having that fix integrated?)

Reply:
No idea, that's old code, I created #673 to track this.